File: harness/schemas/action_receipt.py
"""Universal Action Receipt schema (JSON-serializable)."""

from __future__ import annotations

from typing import Any, Literal, TypedDict

UniversalActionDecision = Literal["ALLOW", "BLOCK", "READ_ONLY", "REQUIRE_APPROVAL"]


class UniversalActionReceipt(TypedDict):
    receipt_id: str
    timestamp: str
    agent: dict[str, Any]
    intent: str
    proposed_action: dict[str, Any]
    policy_evaluated: dict[str, str]
    decision: UniversalActionDecision
    execution_state: dict[str, str]
    evidence_bundle_hash: str

File: nexus-shield-dashboard/types/action-receipt.ts
export type UniversalActionDecision = 'ALLOW' | 'BLOCK' | 'READ_ONLY' | 'REQUIRE_APPROVAL';

export interface UniversalActionReceipt {
  receipt_id: string;
  timestamp: string;
  agent: {
    id: string;
    identity_verified: boolean;
  };
  intent: string;
  proposed_action: {
    tool: string;
    params: Record<string, unknown>;
  };
  policy_evaluated: {
    rule_id: string;
    action: string;
  };
  decision: UniversalActionDecision;
  execution_state: {
    before_hash: string;
    after_hash: string;
  };
  evidence_bundle_hash: string;
}
